src/client/src/features/ValidationError.js

```diff
@@ -0,0 +1,25 @@
+export default class ValidationError {
+  constructor(type, description) {
+    this.type = type;
+    this.description = description;
+  }
+
+  static Validate(data) {
+    const emailRegex = /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/;
+    const validationErrors = [];
+    if (!emailRegex.test(data.email)) validationErrors.push(new ValidationError(
+      'email', 'Invalid email',
+    ));
+    if (data.password.length < 8) validationErrors.push(new ValidationError(
```

Comment:
maybe add `    if (!data.password) {
      validationErrors.push(new ValidationError('password', 'Password is required'));`